flask/app/gene_viewer.py

```diff
@@ -0,0 +1,248 @@
+                == models.users_groups_table.columns.group_id,
+            )
+            .options(
+                contains_eager(models.Gene.variants)
+                .contains_eager(models.Variant.genotype)
+                .contains_eager(models.Genotype.analysis)
+                .contains_eager(models.Analysis.datasets)
+                .contains_eager(models.Dataset.tissue_sample)
+                .contains_eager(models.TissueSample.participant)
+                .contains_eager(models.Participant.family),
+            )
+            .filter(or_(*filters), models.users_groups_table.columns.user_id == user_id)
+        )
+
+    else:
+        app.logger.debug("Processing query - unrestricted based on user id.")
+        query = (
+            models.Gene.query.join(models.Gene.variants)
+            .join(models.Variant.genotype)
+            .join(models.Genotype.analysis, models.Genotype.dataset)
+            .join(models.Dataset.tissue_sample)
+            .join(models.TissueSample.participant)
+            .join(models.Participant.family)
+            .options(
+                contains_eager(models.Gene.variants)
+                .contains_eager(models.Variant.genotype)
+                .contains_eager(models.Genotype.analysis)
+                .contains_eager(models.Analysis.datasets)
+                .contains_eager(models.Dataset.tissue_sample)
+                .contains_eager(models.TissueSample.participant)
+                .contains_eager(models.Participant.family),
+            )
+            .filter(or_(*filters))
+        )
+
+    q = query.all()
+
+    if len(q) == 0:
+        app.logger.error("No requested genes were found.")
+        abort(400, description="No requested genes were found.")
+    elif len(q) < len(genes):
+        app.logger.error("Not all requested genes were found.")
+        abort(400, description="Not all requested genes were found.")
+
+    if request.accept_mimetypes["application/json"]:
+
+        app.logger.info("Application/json Accept header requested")
+        return jsonify(
+            [
+                {
+                    **asdict(gene),
+                    "variants": [
+                        {
+                            **asdict(variant),
+                            "zygosity": [
+                                genotype.zygosity for genotype in variant.genotype
+                            ],
+                            "alt_depths": [
+                                genotype.alt_depths for genotype in variant.genotype
+                            ],
+                            "burden": [
+                                genotype.burden for genotype in variant.genotype
+                            ],
+                            "participant_codenames": [
+                                dataset.tissue_sample.participant.participant_codename
+                                for dataset in variant.analysis.datasets
+                            ],
```

Comment:
I think you guys may have already figured out a solution, but one approach might be is something like:
```
variants: [
    { 
      variant_id: 1,
      ...,
      genotypes: [
         {
            dataset_id: 1,
            zygosity: ...
         }
      ] 
    }
  ]

```

Reply:
this was actually super helpful to visualize, thanks! see: a0326d3 and the output in the edited initial PR post. looking back at it now instead of serializing all of genotype i guess i could explicitly only return the genotype fields - not sure if it matters too much though